--- a/include/target.mk
+++ b/include/target.mk
@@ -18,6 +18,10 @@
 # @CONFIG_* values.
 ##
 DEFAULT_PACKAGES:=\
+	luci-app-advancedplus luci-app-firewall luci-app-package-manager \
+	luci-app-wizard luci-base luci-compat luci-lib-ipkg luci-lib-fs luci-app-log-viewer \
+	luci-app-argon-config luci-app-ddns-go luci-app-openclash luci-app-adguardhome tcpdump-mini open-vm-tools \
+	wget-ssl curl autocore htop nano zram-swap kmod-lib-zstd kmod-tcp-bbr bash openssh-sftp-server block-mount resolveip ds-lite swconfig luci-app-fan luci-app-filemanager luci-app-ipsec \
 	base-files \
 	ca-bundle \
 	dropbear \
@@ -47,10 +51,10 @@ DEFAULT_PACKAGES.nas:=\
 # @brief Default packages for @DEVICE_TYPE router.
 ##
 DEFAULT_PACKAGES.router:=\
-	dnsmasq \
-	firewall4 \
-	nftables \
-	kmod-nft-offload \
+	dnsmasq-full \
+	firewall \
+	kmod-ipt-nat \
+	kmod-ipt-nat6 \
 	odhcp6c \
 	odhcpd-ipv6only \
 	ppp \
