--- a/include/feeds.mk
+++ b/include/feeds.mk
@@ -34,7 +34,7 @@
 # 1: destination file
 define FeedSourcesAppendOPKG
 ( \
-  echo 'src/gz %d_core %U/targets/%S/packages'; \
+  echo 'src/gz %d_core %U/targets/%S/$(LINUX_VERSION)'; \
   $(strip $(if $(CONFIG_PER_FEED_REPO), \
 	echo 'src/gz %d_base %U/packages/%A/base'; \
 	$(if $(CONFIG_BUILDBOT), \
@@ -53,4 +53,4 @@ define FeedSourcesAppendAPK
-  echo '%U/targets/%S/packages/packages.adb'; \
+  echo '%U/targets/%S/$(LINUX_VERSION)/packages.adb'; \
   $(strip $(if $(CONFIG_PER_FEED_REPO), \
 	echo '%U/packages/%A/base/packages.adb'; \
 	$(if $(CONFIG_BUILDBOT), \
