--- a/target/linux/stm32/modules.mk
+++ b/target/linux/stm32/modules.mk
@@ -64,7 +64,7 @@
   KCONFIG:=CONFIG_VIDEO_STM32_DCMI
   FILES:=$(LINUX_DIR)/drivers/media/platform/st/stm32/stm32-dcmi.ko
   AUTOLOAD:=$(call AutoProbe,stm32-dcmi)
-  $(call AddDepends/video,@TARGET_stm32 +kmod-video-videobuf2 +kmod-video-dma-contig +kmod-video-async +kmod-video-fwnode)
+  $(call AddDepends/video,@TARGET_stm32 +kmod-video-videobuf2 +kmod-video-dma-contig +LINUX_6_12:kmod-video-async +LINUX_6_12:kmod-video-fwnode)
 endef
 
 $(eval $(call KernelPackage,stm32-dcmi))



--- a/package/kernel/linux/modules/video.mk
+++ b/package/kernel/linux/modules/video.mk
@@ -628,7 +628,7 @@
 define KernelPackage/video-core
   SUBMENU:=$(VIDEO_MENU)
   TITLE=Video4Linux support
-  DEPENDS:=+PACKAGE_kmod-i2c-core:kmod-i2c-core +kmod-media-controller
+  DEPENDS:=+PACKAGE_kmod-i2c-core:kmod-i2c-core +LINUX_6_12:kmod-media-controller
   KCONFIG:= \
 	CONFIG_MEDIA_CAMERA_SUPPORT=y \
 	CONFIG_VIDEO_DEV \
